public/app/features/search/components/SearchResults.tsx

```diff
@@ -0,0 +1,159 @@
+        onFolderExpanding();
+      }
+
+      section.toggle(section).then(() => {
+        if (onSelectionChanged) {
+          onSelectionChanged();
+        }
+      });
+    }
+  };
+
+  // TODO display 'No results' messages after manage dashboards is refactored
+  if (!results) {
+    return null;
+  }
+
+  return (
+    <ul className={styles.wrapper}>
+      {results.map(section => (
+        <li aria-label="Search section" className={styles.section} key={section.title}>
+          <SectionHeader onSectionClick={toggleFolderExpand} {...{ onToggleSelection, editable, section }} />
+          <ul aria-label="Search items" className={styles.wrapper}>
+            {section.expanded &&
+              section.items.map(item => (
+                <SearchItem key={item.id} {...{ item, editable, onToggleSelection, onTagSelected }} />
+              ))}
+          </ul>
+        </li>
+      ))}
+    </ul>
+  );
+};
+
+const getSectionStyles = (theme: GrafanaTheme) => {
+  return {
+    wrapper: css`
+      list-style: none;
+    `,
+    section: css`
+      background: ${theme.colors.panelBg};
+      border-bottom: solid 1px ${theme.isLight ? theme.colors.gray95 : theme.colors.gray25};
+      padding: 0px 4px 4px 4px;
+      margin-bottom: 3px;
+    `,
+  };
+};
+
+interface SectionHeaderProps {
+  section: DashboardSection;
+  onSectionClick: (section: DashboardSection) => void;
+  onToggleSelection: ItemClickWithEvent;
+  editable: boolean;
+}
+
+const SectionHeader: FC<SectionHeaderProps> = ({ section, onSectionClick, onToggleSelection, editable }) => {
+  const theme = useTheme();
+  const styles = getSectionHeaderStyles(theme, section.selected);
+
+  const expandSection = () => {
+    onSectionClick(section);
+  };
+
+  return !section.hideHeader ? (
+    <div className={styles.wrapper} onClick={expandSection}>
+      <SearchCheckbox
+        editable={editable}
+        checked={section.checked}
+        onClick={(e: MouseEvent) => onToggleSelection(section, e)}
+      />
+      <Icon className={styles.icon} name={section.icon as IconType} />
+
+      <span className={styles.text}>{section.title}</span>
+      {section.url && (
+        <a href={section.url} className={styles.link}>
+          <Icon name="cog" />
+        </a>
+      )}
+      <Icon name={section.expanded ? 'angle-down' : 'angle-right'} className={styles.toggle} />
+    </div>
+  ) : (
+    <div className={styles.wrapper} />
+  );
+};
+
+const getSectionHeaderStyles = (theme: GrafanaTheme, selected = false) => {
+  const { sm, xs } = theme.spacing;
+  return {
+    wrapper: cx(
+      css`
+        display: flex;
+        font-size: ${theme.typography.size.base};
+        padding: ${sm} ${xs} ${xs};
+        color: ${theme.colors.textWeak};
+
+        &:hover,
+        &.selected {
+          color: ${theme.colors.text};
+        }
+
+        &:hover {
+          a {
+            opacity: 1;
+          }
+        }
+      `,
+      'pointer',
+      { selected }
+    ),
```

Comment:
what does this cx function do? `cx(css`some css`, 'pointer', { selected })`  I do not understand this, is there a pointer class? and passing an object with a boolean property as last argument what does that do?

Reply:
Class `pointer` adds cursor pointer on hover, not sure if we have anything like that in the new styles, I also wanted to reuse the existing styles for now. 
`{ selected }` adds a selected class to the element if `selected` prop is true. It's done that way so  hover and selected could be combined because they both have the same styles.  
```  
&:hover,
&.selected {
  color: ${theme.colors.text};
}
```